src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -114,20 +126,49 @@ static bool analyze(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> f
                 return !failedList.Any();
             }
 
+            static bool analyzeSymbols(ElfBinary binary, out List<SymbolEntry<ulong>> failedList)
+            {
+                failedList = new List<SymbolEntry<ulong>>();
+
+                var symbols = binary.ELF.Sections.FirstOrDefault(s => s.Type == SectionType.DynamicSymbolTable) as SymbolTable<ulong>;
+                foreach (SymbolEntry<ulong> symbol in symbols.Entries)
+                {
+                    if (symbol.Name != "__stack_chk_fail" && symbol.Name != "stack_chk_guard" && symbol.Name != "__intel_security_cookie")
+                    {
+                        failedList.Add(symbol);
+                    }
+                    else
+                    {
+                        return true;
+                    }
+                }
+                return !failedList.Any();
+            }
+
             if (binary is ElfBinary elf)
             {
-                if (!analyze(elf, out failedList))
+                if (!analyzeDwarf(elf, out failedList))
                 {
-                    // The stack protector was not found in '{0}'.
-                    // This may be because '--stack-protector-strong' was not used,
-                    // or because it was explicitly disabled by '-fno-stack-protectors'.
-                    // Modules did not meet the criteria: {1}
-                    context.Logger.Log(this,
-                        RuleUtilities.BuildResult(FailureLevel.Error, context, null,
-                            nameof(RuleResources.BA3003_Error),
-                            context.TargetUri.GetFileName(),
-                            DwarfUtility.GetDistinctNames(failedList, context.TargetUri.GetFileName())));
-                    return;
+                    // Analysis using DWARF info failed.
+                    // This could be because the binary simply doesn't have DWARF info.
+                    // So we'll fall back to a symbol search similar to checksec, just to be sure.
+                    List<SymbolEntry<ulong>> failedList_symbols;
+                    if (!analyzeSymbols(elf, out failedList_symbols))
```

Comment:
>failedList_symbols [](http://example.com/codeflow?start=49&length=18)

Yes, I see no utilization of this data, so looks like we can drop it. Let's move these static local functions into standard C# methods as well. Unless someone can clarify the specific technical value of embedding them locally, of course. :)